netbox/core/api/serializers_/tasks.py

```diff
@@ -0,0 +1,84 @@
+from rest_framework import serializers
+
+__all__ = (
+    'BackgroundTaskSerializer',
+    'BackgroundQueueSerializer',
+    'BackgroundWorkerSerializer',
+)
+
+
+class BackgroundTaskSerializer(serializers.Serializer):
+    id = serializers.CharField()
+    url = serializers.HyperlinkedIdentityField(
+        view_name='core-api:rqtask-detail',
+        lookup_field='id',
+        lookup_url_kwarg='pk'
+    )
+    description = serializers.CharField()
+    origin = serializers.CharField()
+    func_name = serializers.CharField()
+    args = serializers.ListField(child=serializers.CharField())
+    kwargs = serializers.DictField()
+    result = serializers.CharField()
+    timeout = serializers.IntegerField()
+    result_ttl = serializers.IntegerField()
+    created_at = serializers.DateTimeField()
+    enqueued_at = serializers.DateTimeField()
+    started_at = serializers.DateTimeField()
+    ended_at = serializers.DateTimeField()
+    worker_name = serializers.CharField()
+    position = serializers.SerializerMethodField()
+    status = serializers.SerializerMethodField()
+    meta = serializers.DictField()
+    last_heartbeat = serializers.CharField()
+
+    is_finished = serializers.BooleanField()
+    is_queued = serializers.BooleanField()
+    is_failed = serializers.BooleanField()
+    is_started = serializers.BooleanField()
+    is_deferred = serializers.BooleanField()
+    is_canceled = serializers.BooleanField()
+    is_scheduled = serializers.BooleanField()
+    is_stopped = serializers.BooleanField()
+
+    def get_position(self, obj) -> int:
+        return obj.get_position()
+
+    def get_status(self, obj) -> str:
+        return obj.get_status()
+
+
+class BackgroundQueueSerializer(serializers.Serializer):
+    name = serializers.CharField()
+    # Note: Queue names are not url safe so not returning a url field
```

Comment:
Is this because the queue names are user-defined and thus potentially contain non-alphanumeric characters? Seems like a fairly low risk that we can mitigate with URL encoding. Ultimately we need to provide a detail endpoint to retrieve a specific RQ queue.